### ISSUE TITLE
Composites flag: calibration plan before flipping default-on

### ISSUE THREAD

**Issue by lilith** [imazen/zenanalyze#6]
## Background

`composites` cargo feature shipped in 0.1.0 (off by default). Gates: `TextLikelihood`, `ScreenContentLikelihood`, `NaturalLikelihood`, `LineArtScore`. The raw signals these consume (variance, edge density, distinct color bins, luma histogram entropy, flat-color block ratio, etc.) are stable in 0.1.x — but the **combinator coefficients** are calibration-driven and known to drift as the corpus matures. That's why the flag is off by default.

When can the flag be flipped to default-on (or even removed entirely)?

## Current state — what the 0.1.0 coefficients actually do

From `zenanalyze/docs/calibration-corpus-2026-04-27.md` and tier3.rs:

```rust
text_likelihood = (entropy_low * 0.4 + edge_hi * 0.3 + chroma_lo * 0.3).clamp(0, 1)
screen_content_likelihood = (flat_high * 0.6 + palette_small * 0.3 + chroma_lo * 0.1).clamp(0, 1)
natural_likelihood = (entropy_hi * 0.3 + palette_large * 0.25 + chroma_moderate * 0.2 + not_flat * 0.25).clamp(0, 1)
line_art_score = min(bimodality, top2_coverage, low_entropy)  // conservative AND-style combinator
```

Empirical performance on the 219-image labeled corpus:

| feature | empirical max | recommended threshold | F1 | AUC |
|---|---:|---|---:|---:|
| TextLikelihood | 0.71 | ≥ 0.30 | 0.682 | 0.774 |
| ScreenContentLikelihood | 0.70 | ≥ 0.60 | 0.750 | 0.831 |
| NaturalLikelihood | 0.69 | ≥ 0.06 | 0.924 | 0.814 |
| `flat_color_block_ratio` (raw) | — | ≥ 0.53 | 0.750 | 0.838 |
| `patch_fraction` (raw) | — | — | — | **0.88** |

Note: `flat_color_block_ratio` raw matches `screen_content_likelihood` F1 (0.750), and `patch_fraction` outperforms it (AUC 0.88 vs 0.83). The composites are weaker than at least one underlying signal — that's the calibration story we need to fix.

## What would unblock default-on

1. **Wider corpus.** 219 images is small; partition stability is unknown. Need ≥ 1000 images covering screen / natural / line-art / mixed at minimum, with held-out validation set.
2. **Re-fit coefficients** using a real optimizer (logistic regression / GBT / shallow MLP) instead of hand-picked weights. Targets:
   - TextLikelihood AUC ≥ 0.85 (currently 0.77)
   - ScreenContentLikelihood AUC ≥ 0.90 (currently 0.83)
   - NaturalLikelihood F1 ≥ 0.95 (currently 0.92)
3. **Stability**: train/test partitions should not move thresholds by more than ±0.05.
4. **Cross-corpus generalization**: no single dataset (CID22 / CLIC / gb82-sc / ProRAW HDR / etc.) should dominate any feature's distribution.

## Coefficient migration prerequisite

Before re-fitting can happen, `coefficient` (the calibration crate) needs to consume the published `zenanalyze` 0.1.0 instead of vendoring its own `evalchroma_ext.rs` copy. Tracked in this session as task #70 — separate issue worth filing if not already.

## Coordinate with #1 and #2

Issue #1 proposes `patch_fraction_structured` and a faster fingerprint, which should land **before** the re-fit so the next composites generation can include those higher-AUC signals. Issue #2 is a 0.2 release scope — composites coefficient bump probably ships in 0.2 alongside the per-pixel cost levers.

## Acceptance

- [ ] coefficient depends on zenanalyze published crate (no vendored copy).
- [ ] Corpus ≥ 1000 images, balanced across content classes, with held-out 20 % validation.
- [ ] All four likelihoods hit the AUC / F1 targets above on validation.
- [ ] Threshold stability ≤ ±0.05 across train/val splits.
- [ ] One follow-up: decide whether to flip `composites` default-on, or to keep it opt-in but document the coefficients as locked-stable for 0.2.x.

**Comment by lilith** [comment]
## Calibration plan

### Goal

"Calibrated" = each likelihood beats its strongest single underlying raw signal on a held-out split, on a corpus larger than 219 images. Quantitative gates:

- TextLikelihood AUC ≥ 0.85 (currently 0.713 — note: the issue body's 0.774 is the pre-stretch number; the docstring at `tier3.rs:1520` lists 0.713 post-stretch)
- ScreenContentLikelihood AUC ≥ 0.90 *and* ≥ `patch_fraction` raw (currently 0.845 with the `experimental` formula, vs `patch_fraction` alone at 0.880)
- NaturalLikelihood F1 ≥ 0.95 at recommended threshold (currently 0.923)
- LineArtScore: needs an AUC measurement at all — none reported in `tier3.rs::HStats` doc (`min(bimodality, top2_coverage, low_entropy)` is unvalidated)
- Stability: thresholds shift by ≤ ±0.05 across train/val splits

### Test set

219-image corpus exists at `docs/calibration-corpus-2026-04-27.md` (375 lines) sourced from `coefficient/benchmarks/classifier-eval/labels.tsv`. Spans 9 sub-corpora (cid22, clic2025-1024, gb82, gb82-sc, imageflow, kadid10k, qoi-benchmark, corpus). Class-imbalanced: photo=174, screen=36, illustration=9 — too few screen exemplars to cross-validate stably.

**Stand up before any flag flip:** expand to ≥ 1000 labeled images with ≥ 200 per class (screen / natural / line-art / mixed), 80/20 train/val split.

### Calibration procedure

1. **Re-fit** each composite via logistic regression on the training half, regressing `is_<class>` ground truth against the *currently used* leaf features (`luma_histogram_entropy`, `edge_density`, `chroma_sh`, `flat_color_block_ratio`, `distinct_color_bins`, `patch_fraction`). Compare against gradient-boosted trees as a non-linear ceiling — if GBT beats LR by > 5 AUC points on validation, the linear combinator is the wrong model.
2. **Evaluate** AUC on the held-out half. Reject any composite that fails to beat its best constituent raw feature by ≥ 0.02 AUC — at that point the composite is pure cost.
3. **Threshold-sweep** validation predictions for max F1, log P/R/F1 at sweep ±0.05 around the optimum to confirm flat plateau (not knife-edge).
4. **Document** new coefficients in CHANGELOG with `[Unreleased] → Changed` entry: old vs new weights, train/val AUC, F1, threshold, corpus version.
5. **Update** docstring p10/p50/p90 ranges in `feature.rs` and the empirical-max constants (`TEXT_MAX`, `SCREEN_MAX`, `NATURAL_MAX` at `tier3.rs:1545`) from the new corpus.

### Default-on gate

Flip only when **all four** of:

- Each likelihood AUC ≥ 0.80 on held-out validation
- Each likelihood ≥ best constituent raw feature AUC by ≥ 0.02
- Threshold stability ≤ ±0.05 across 5-fold CV
- Corpus ≥ 1000 images with balanced class coverage

### Alternative — drop the feature

Issue body already shows `patch_fraction` (AUC 0.88) > `screen_content_likelihood` (0.85), and `flat_color_block_ratio` (0.838) ≈ `screen_content_likelihood` (F1 tied at 0.750). If re-fitting on the larger corpus doesn't lift composites ≥ 0.02 AUC over their best constituent, **delete the `composites` feature entirely** and document the raw features as the recommended consumer signals. The per-call cost is small but non-zero, and one-feature-better-than-the-composite is the wrong architecture to ship default-on.

### Prerequisites

- `coefficient` crate must consume published `zenanalyze` 0.1.x (not vendored `evalchroma_ext.rs`) before re-fit — already noted in issue body, file as separate tracking issue if not present.
- Issue #1 (`patch_fraction_structured`) should land first so the re-fit can include it.

**Comment by lilith** [comment]
Closing — verified stale/resolved on the current trunk.

**Summary:** The composites cargo feature and its four likelihood composites were deleted in b1623ba (ids 27/28/29/45 reserved), following the issue's own 'drop the feature' alternative since raw signals like patch_fraction outperformed them. There is no flag left to calibrate or flip.

**Evidence (checked against `origin` trunk on 2026-08-27/28):**
b1623ba (2026-05-01) 'cull 17 features + new hdr cargo flag, delete composites' is on origin/main; no `composites` feature in origin/main Cargo.toml; src/feature.rs:2580-2583 reserves ids 27/28/29/45 for the removed TextLikelihood / ScreenContentLikelihood / NaturalLikelihood / LineArtScore; the names survive only in doc comments as 'retired' (feature.rs:433, :657, :685, :1831). This is the 'Alternative — drop the feature' path from the issue's own calibration-plan comment.

**References:** `b1623ba`; files: `feature.rs:433`, `src/feature.rs:2580`

---
*This issue was reviewed and closed by Claude (Anthropic) operating under the repo owner's account, as part of the 2026-08-26/27 cross-repo issue triage: a single-pass review classified it as likely stale, and an independent read-only verification then tried to refute that against the trunk and could not. If any part of the original ask is still open, reopen with the specific item — nothing in this comment was generated without opening the cited files.*
